---
title: 'Style - No Lead Zero'
layout: base
permalink: '/rules/style/noLeadZero/index.html'
anchors:
  - ''
---

# No Lead Zero

The `noLeadZero` rule will eliminate leading zeros from numbers expressed within values.

---

<!--

üôå - Recommended Choice
üëç - Good Choice
üëé - Not Recommended
ü§° - Clown Choice
üò≥ - Bad Choice

-->

::: rule üëç

#### false

:::

Lorem Ipsum

<!-- RULES ARE REQUIRED -->

```json:rules
{
  "language": "css",
  "style": {
    "noLeadZero": false
  }
}
```

<!-- prettier-ignore -->
```css
.class-a {
  width: 0.5px;
  height: 0.05rem;
  left: 0.1em;
  right: 0.50cm;
  top: 0.3ch;
}
```

---

::: rule üëç

#### true

:::

```json:rules
{
  "language": "css",
  "style": {
    "noLeadZero": true
  }
}
```

<!-- prettier-ignore -->
```css
.class-a {
  width: 0.5px;
  height: 0.05rem;
  left: 0.1em;
  right: 0.50cm;
  top: 0.3ch;
}
```
