---
title: 'Style - Sort Properties'
layout: base
permalink: '/rules/style/sortProperties/index.html'
anchors:
  - ''
---

# Sort Properties

The `sortProperties` rule will apply alphabetical sorting to style class selector properties..

---

<!--

üôå - Recommended Choice
üëç - Good Choice
üëé - Not Recommended
ü§° - Clown Choice
üò≥ - Bad Choice

-->

::: rule üôå

#### true

:::

Lorem Ipsum

<!-- RULES ARE REQUIRED -->

```json:rules
{
  "language": "css",
  "style": {
    "sortProperties": true
  }
}
```

<!-- prettier-ignore -->
```css
.class {
  z-index: 999;
  color: #fff;
  width: 200px;
  background-repeat: no-repeat;
  background-position: top;
  background-attachment: fixed;
  font-weight: 100;
  font-style: bold;
  margin-top: 100px;
  display: flex;
  position: absolute;
  float: right;
  margin-left: 100px;
  padding-right: 25px;
  transition: ease-in;
  visibility: visible;
  padding-bottom: 40px;
  font-weight: 100;
  font-style: bold;
  padding-left: 25px;
  visibility: collapse;
  min-inline-size: inherit;
}
```

---

::: rule üëç

#### false

:::

```json:rules
{
  "language": "css",
  "style": {
    "sortProperties": false
  }
}
```

<!-- prettier-ignore -->
```css
.class {
  z-index: 999;
  color: #fff;
  width: 200px;
  background-repeat: no-repeat;
  background-position: top;
  background-attachment: fixed;
  font-weight: 100;
  font-style: bold;
  margin-top: 100px;
  display: flex;
  position: absolute;
  float: right;
  margin-left: 100px;
  padding-right: 25px;
  transition: ease-in;
  visibility: visible;
  padding-bottom: 40px;
  font-weight: 100;
  font-style: bold;
  padding-left: 25px;
  visibility: collapse;
  min-inline-size: inherit;
}
```
