---
title: 'LANGUAGE - RULE_NAME'
layout: base
permalink: '/rules/[LANGUAGE]/[RULE_NAME]/index.html'
anchors:
  - ''
---

# RULE_NAME

Lorem Ipsum

::: note
Lorem Ipsum
:::

##### RELATED_RULES

Lorem Ipsum

---

# Rule Options

This is a LANGUAGE_NAME specific formatting rule which defaults to using `preserve` when no option has been specified.

<!--

üôå - Recommended Choice
üëç - Good Choice
üëé - Not Recommended
ü§° - Clown Choice
üò≥ - Bad Choice

-->

::: rule üëç

#### RULE_OPTION

:::

Lorem Ipsum

<!-- RULES ARE REQUIRED -->

```json:rules
{
  "language": "",
}
```

<!-- prettier-ignore -->
```html

```

---
